homeassistant/components/light/vera.py

```diff
@@ -0,0 +1,181 @@
+
+
+name
+*Optional
+This parameter allows you to override the name of your Vera device in the HA
+interface, if not specified the value configured for the device in your Vera
+will be used
+
+
+exclude
+*Optional
+This parameter allows you to exclude the specified device from homeassistant,
+it should be set to "true" if you want this device excluded
+
+"""
+import logging
+import time
+
+from homeassistant.helpers import ToggleDevice
+# pylint: disable=no-name-in-module, import-error
+import homeassistant.external.vera.vera as veraApi
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """ Find and return Vera lights. """
+    try:
+        base_url = config.get('vera_controller_url')
+        if not base_url:
+            _LOGGER.error(
+                "The required parameter 'vera_controller_url'"
+                " was not found in config"
+            )
+            return False
+
+        device_data = config.get('device_data', None)
+
+        controller = veraApi.VeraController(base_url)
+        devices = controller.get_devices('Switch')
+
+        lights = []
+        for device in devices:
+            extra_data = get_extra_device_data(device_data, device.deviceId)
+            exclude = False
+            if extra_data:
+                exclude = extra_data.get('exclude', False)
+
+            if exclude is not True:
+                lights.append(VeraLight(device, extra_data))
+
+        add_devices_callback(lights)
+    # pylint: disable=broad-except
+    except Exception as inst:
+        _LOGGER.error("Could not find Vera lights: %s", inst)
+        return False
+
+
+def get_extra_device_data(device_data, device_id):
+    """ Gets the additional configuration data by Vera device Id """
+    if not device_data:
+        return None
+
+    for item in device_data:
+        if item.get('vera_id') == device_id:
+            return item
+
+    return None
+
+
+class VeraLight(ToggleDevice):
+    """ Represents a Vera light """
+    is_on_status = False
+    # for debouncing status check after command is sent
+    last_command_send = 0
+    extra_data = None
+
+    def __init__(self, vera_device, extra_data=None):
+        self.vera_device = vera_device
+        self.extra_data = extra_data
+
+    @property
+    def name(self):
+        """ Get the mame of the light. """
+        if self.extra_data and self.extra_data.get('name'):
+            return self.extra_data.get('name')
+        return self.vera_device.name
+
+    @property
+    def state_attributes(self):
+        attr = super().state_attributes
+
+        if self.vera_device.has_battery:
+            attr['Battery'] = self.vera_device.battery_level + '%'
+
+        if self.vera_device.is_armable:
+            armed = self.vera_device.refresh_value('Armed')
+            attr['Armed'] = 'True' if armed == '1' else 'False'
```

Comment:
`Armed` and `Last Tripped` will probably show up in more sensors. Let's add them to `homeassistant.const`.